### PR TITLE
Confidence Score fix to highlight only the entity which is present in the label name

### PR DESCRIPTION
**Pull request by dristysrivastava**
**Issue:** In snippets in Safe Loader app, confidence score was highlighted for all the entities/topics, rather it should only be highlighted for the specific label

**Before**
Screenshot 2024-08-23 at 13 18 08

**After**
Screenshot 2024-08-23 at 13 06 25

### CODE REVIEW

File: pebblo/app/service/doc_helper.py [hunk 1]
```diff
@@ -239,6 +239,30 @@ def _update_app_details(self, raw_data, ai_app_docs):
             loader["sourceFiles"] = new_source_files
         self.app_details["report_metadata"] = raw_data
 
+    @staticmethod
+    def _create_data_source_findings(data_source_findings):
```
Comment (gr8nishan) [inline pebblo/app/service/doc_helper.py:243]: Some comments will be helpful

Comment (dristysrivastava) [reply pebblo/app/service/doc_helper.py:243]: sure

File: pebblo/app/service/doc_helper.py [hunk 2]
```diff
@@ -239,6 +239,30 @@ def _update_app_details(self, raw_data, ai_app_docs):
...
+        for data in data_source_findings:
+            for snippet in data.get("snippets"):
+                entity_details = {}
+                topic_details = {}
+                if snippet.get("entityDetails"):
+                    if data["labelName"] in snippet.get("entityDetails").keys():
```
Comment (shreyas-damle) [inline pebblo/app/service/doc_helper.py:249]: Can we combine these nested if statements?

Comment (dristysrivastava) [reply pebblo/app/service/doc_helper.py]: sure

**Review by gr8nishan — APPROVED** [review]

**Review by shreyas-damle — APPROVED** [review]

**Pull request merged**